scripts/helpers/post_comment.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+
+#
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+import coloredlogs
+import github
+import logging
+import traceback
+import sys
+
+def main():
+  """Main task if executed standalone."""
+  parser = argparse.ArgumentParser(description='Fetch master build artifacts.')
+  parser.add_argument(
+      '--github-api-token',
+      type=str,
+      help='Github API token to upload the report as a comment')
+  parser.add_argument(
+      '--github-repository', type=str, help='Repository to use for PR comments')
+  parser.add_argument(
+      '--pr',
+      type=int,
+      help='The PR number of the pull request')
+  parser.add_argument(
+      '--title',
+      type=str,
+      help='Title of the comment')
+  parser.add_argument(
+      '--log-level',
+      default=logging.INFO,
+      type=lambda x: getattr(logging, x),
+      help='Configure the logging level.')
+  args = parser.parse_args()
+
+  # Ensures somewhat pretty logging of what is going on
+  logging.basicConfig(
+      level=args.log_level,
+      format='%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+  coloredlogs.install()
+
+  if not args.github_api_token:
+    logging.error('Required arguments missing: github api token is required.')
+    return
+
+  try:
+    logging.info('Uploading report to "%s", PR %d', args.github_repository, args.pr)
+    logging.info('Title "%s"', args.title)
+
+    rawText = sys.stdin.read()
+    logging.info('Content "%s"', rawText)
+
+    api = github.Github(args.github_api_token)
+    repo = api.get_repo(args.github_repository)
+    pull = repo.get_pull(args.pr)
+
+    # Remove old comment
+    for comment in pull.get_issue_comments():
+      #The action token doesn't have permission to get user. Ignore user check
+      #if not comment.user.id == api.get_user().id:
+      #  continue
+      if not comment.body.startswith(args.title):
+        continue
+      logging.info('Removing obsolete comment with heading "%s"', (args.title))
```

Comment:
This seems like a pretty sharp knife: if someone accidentally passes in empty string, this will delete all the comments, no?

I'd prefer it if all comments posted by this script had a known prefix from the script itself (which gets prepended to args.title).

Also, this is presumably copied in large part from bloat_check.py.  How feasible would it be to make that script use this one?

Reply:
Let me re-enable the user id check, to see if we can guard this one